mechanicalsoup/browser.py

```diff
@@ -181,7 +181,11 @@ def _request(self, form, url=None, **kwargs):
                     # (or submit button) enctype attribute is set to
                     # "multipart/form-data". We don't care, simplify.
                     if not value:
-                        continue
+                        # create a temporary file for upload
```

Comment:
I'm not sure we need to be making temporary files here. So long as we are passing the `name` to requests, then we will get the multipart form data we want in the response, even if the value is empty.

I'd recommend something like this instead:
```python
                    if value != "": 
                        value = open(value, "rb")
                    # If value is the empty string, we still pass it for consistency
                    # with browsers (see #250).
                    files[name] = value